doc/rtd/introduction.rst

```diff
@@ -0,0 +1,116 @@
+well. By automating routine setup tasks, cloud-init ensures repeatability and
+efficiency in your system provisioning.
+
+What is the benefit of cloud-init?
+==================================
+
+When you deploy a new cloud instance, cloud-init takes an initial configuration
+that you supply, and it automatically applies those settings when the instance
+is created. It’s rather like writing a to-do list, and then letting cloud-init
+deal with that list for you.
+
+The real power of cloud-init comes from the fact that you can re-use your
+configuration instructions as often as you want, and always get consistent,
+reliable results. If you’re a system administrator and you want to deploy a
+whole fleet of machines, you can do so with a fraction of the time and effort
+it would take to manually provision them.
+
+What does cloud-init do?
+========================
+
+Cloud-init can handle a range of tasks that normally happen when a new instance
+is created. It's responsible for activities like setting the hostname,
+configuring network interfaces, creating user accounts, and even running
+scripts. This streamlines the deployment process; your cloud instances will all
+be automatically configured in the same way, which reduces the chance to
+introduce human error.
+
+How does cloud-init work?
+=========================
+
+The operation of cloud-init broadly takes place in two separate phases during
+the boot process. The first phase is during the early (local) boot stage,
+before networking has been enabled. The second is during the late boot stages,
+after cloud-init has applied the networking configuration.
+
+During early boot
+-----------------
+
+In this pre-networking stage, cloud-init discovers the datasource, obtains
+all the configuration data from it, and configures networking. In this phase,
+it will:
+
+* **Identify the datasource**:
+  The hardware is checked for built-in values that will identify the datasource
+  your instance is running on. The datasource is the source of all
+  configuration data.
+* **Fetch the configuration**:
+  Once the datasource is identified, cloud-init fetches the configuration data
+  from it. This data tells cloud-init what actions to take. This can be in the
+  form of:
+
+  * **Metadata** about the instance, such as the machine ID, hostname and
+    network config, or
+  * **Vendor data** and/or **user data**. These take the same form, although
+    Vendor data is provided by the cloud vendor, and user data is provided by
+    the user. These data are usually applied in the post-networking phase, and
+    might include:
+
+    * Hardware optimisations
+    * Integration with the specific cloud platform
+    * SSH keys
+    * Custom scripts
+
+* **Apply network configuration**:
+  Cloud-init sets up network interfaces, assigns IP and MAC addresses, and
+  configures DNS.
+
+During late boot
+----------------
+
+In the boot stages that come after the network has been configured, cloud-init
```

Comment:
Nit: This sentence feels a little complex. Can we separate the `but` clause into a separate sentence?

Reply:
Absolutely!